src/apps/Permissions/Permissions.js

```diff
@@ -64,29 +75,35 @@ function Permissions({
 
   const [homeTab, setHomeTab] = useState(0)
 
-  // `params` should change every time we navigate into and out of a detailed
+  // `localPath` should change every time we navigate into and out of a detailed
   // permissions view, so this ensures the user starts at the top of the screen
   // on every navigation change
   useEffect(() => {
     scrollTopElement.current.scrollIntoView()
-  }, [params])
+  }, [localPath])
+
+  const location = getLocation(localPath, apps)
 
   const openHome = useCallback(() => {
-    onParamsRequest(null)
-  }, [onParamsRequest])
+    onPathRequest('/')
+  }, [onPathRequest])
 
   const openApp = useCallback(
     proxyAddress => {
-      onParamsRequest(`app.${proxyAddress}`)
+      onPathRequest(`/app/${proxyAddress}`)
     },
-    [onParamsRequest]
+    [onPathRequest]
   )
 
   const manageRole = useCallback(
     (proxyAddress, roleBytes) => {
-      onParamsRequest(`app.${proxyAddress}.role.${roleBytes}`)
+      onPathRequest(
+        location.screen === 'app'
+          ? `/app/${proxyAddress}/role/${roleBytes}`
+          : `/role/${proxyAddress}${roleBytes}`
```

Comment:
What do you think about using a `.` to separate the two, to make it a bit nicer?

Reply:
Yes let’s do that :+1: After the routing system has been a bit reshaped, I think we should make it part of the search (`?=manageRole=…`), since it doesn’t belong to `/app/…` anymore (and we could shorten the address / roleBytes).